integrationtest/mapper/pojo/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/IndexedEmbeddedBaseIT.java

```diff
@@ -657,18 +657,12 @@ public String getInvalid() {
 			}
 		}
 
-		backendMock.expectSchema( INDEX_NAME, b -> b
-				.field( "text", String.class )
-				// TODO HSEARCH-899 this isn't supposed to be legal => throw an exception for
-				.objectField( "invalid", c -> {
-				} )
-		);
-
-		JavaBeanMapping mapping = setupHelper.start()
+		SubTest.expectException( () -> setupHelper.start()
 				.withAnnotatedEntityTypes( IndexedEntity.class )
-				.setup();
-
-		backendMock.verifyExpectationsMet();
+				.setup() )
+				.assertThrown()
+				.isInstanceOf( SearchException.class )
+				.hasMessageContaining( "HSEARCH700060: The indexed embedded property 'invalid' of type 'java.lang.String' does not provide any valid field." );
```

Comment:
Please don't mention the error code ("HSEARCH700060") in these assertions, they may change before the release (upon merging the PR, for example) and to be honest we don't really care about the exact code.

Reply:
Also, you'll probably want to check the context of the error (type, property name) in the failure report. See how we define expectations in `error_includePaths_nonMatched`.